### PR TITLE
feat: add --prettier-path cli option

### PR DESCRIPTION
**Pull request by shannonmoeller**
A stab at implementing #12.

### CODE REVIEW

**Comment by mikew** [comment]
Awesome! Looks pretty good so far but I'll give a deeper look soon.

File: src/linter.ts [hunk 1]
```diff
@@ -110,12 +113,16 @@ export const invoke = (
 ) => {
   process.chdir(cwd)
 
-  let cache = prettierCache.get(cwd)
+  const parsedOptions = parseArguments(args)
+  const prettierPath = parsedOptions.prettierPath
+  const cacheKey = `${cwd};${parsedOptions.prettierPath || ''}`
```
Comment (mikew) [inline src/linter.ts:117]: Any reasoning behind including the prettier path in the cache key? I'm not against it, just curious on your thoughts behind it.

Comment (shannonmoeller) [reply src/linter.ts:117]: The thought was to ensure there aren't cache collisions when people run `prettier_d_slim` in the same directory with different `--prettier-path` values. Wasn't entirely sure it mattered, but seemed like a good idea.

File: src/linter.ts [hunk 2]
```diff
@@ -8,13 +8,14 @@ import resolve from 'resolve'
...
 
-function createCache(cwd: string) {
-  let prettierPath
-  try {
-    prettierPath = resolve.sync('prettier', { basedir: cwd })
-  } catch (e) {
-    // module not found
-    prettierPath = resolve.sync('prettier')
+function createCache(cacheKey: string, cwd: string, prettierPath?: string) {
+  if (prettierPath == null) {
```
Comment (shannonmoeller) [inline src/linter.ts:12]: Hmm, when the path is provided, we should probably run resolve with the cwd to support relative paths. I think this currently only works with fully resolved paths right now.

File: src/linter.ts (hunk 1 above)
Comment (mikew) [reply src/linter.ts:117]: 👍 kinda figured!

File: src/linter.ts (hunk 2 above)
Comment (mikew) [reply src/linter.ts:12]: Hm, I'm not totally worried about that, I'm thinking if a person is using `--prettier-path`, they can easily supply a full path (via `--prettier-path="${PWD}/.yarn/sdks/"`.

Is there no `yarn sdk prettier` command that dumps out a full path? I don't much use yarn in personal projects, and at work we use yarn v1, and the two versions can't coexist.

If anything, we could add an `else` below, and only resolve with the cwd if the given `prettierPath` doesn't start with a `/`.

Comment (shannonmoeller) [reply src/linter.ts:12]: That's fair. It's easy enough to add later and a backwards compatible change if the need arises.

**Comment by mikew** [comment]
Thanks for this! Noticed one small issue, the `prettier-path` option was added to the `boolean` list of CLI flags, which ended up breaking everything 😅. Removing it from there got everything working as expected. Also helped to write some kind of test suite. I'll merge this in and get it released ASAP.

**Comment by shannonmoeller** [comment]
Whoops! Sorry about that. Thanks for fixing and merging!

**Pull request merged**